### DISCUSSION TITLE
Get preset glide transforms from the content API with REST

### DISCUSSION THREAD

**Discussion by ewhicher** [statamic/cms#5756] [Troubleshooting] [answered]
Is there a way to get glide preset transforms back from the assets endpoint (or any other endpoint) of the Content API?

For bonus points I'd also love to get the glide secret so I can do asset transforms on the fly while keeping the `secure` flag `true`... but that might be a pip dream!

**Comment by jasonvarga** [comment, marked as answer]
You can customize the resources.
https://statamic.dev/rest-api#customizing-resources

In one of these resource classes, you could have it output a Glide URL.

The GraphQL docs has an example of doing that. You could take the code and put it into the REST API Resource class.
https://statamic.dev/graphql#custom-fields

**Comment by afonic** [comment]
Extend the class, for example create app/Asset.php:
```
<?php

namespace App;

use Statamic\Facades\Image;

class Asset extends \Statamic\Assets\Asset
{
    public function toShallowAugmentedCollection($keys = null)
    {	
    	return parent::toShallowAugmentedCollection()->merge($this->getPresetImages());
    }

    protected function getPresetImages()
    {
    	$presets = config('statamic.assets.image_manipulation.presets');
    	$manipulator = Image::manipulate($this);
		$presetSizes = [];
		foreach ($presets as $preset => $value) {
			$manipulator->p($preset);
			$presetSizes[$preset] = $manipulator->build();
		}
		return $presetSizes;
    }
}
```

In your service provider:
```
public function boot()
    {
        $this->app->bind(\Statamic\Contracts\Assets\Asset::class, function () {
            return new \App\Asset;
        });
    }
```

You need to see which methods you need to override, in this example it helps us have the presets in the Content API response of a Collection.

**Comment by ewhicher** [comment]
@jasonvarga your answer worked great, I hadn't read the Graph QL section of the docs. Sadly, however, I need it on the `assets` endpoint, not per individual asset. Any suggestions?

@afonic I tried your example but nothing was coming through...

**Reply by jasonvarga** [reply]
The assets endpoint just returns multiple asset resources. Did you try it?

**Reply by ewhicher** [reply]
...I did...multiple times....in the browser....it was showing cached results 🤦‍♂️

This is working perfectly now, thanks!

**Reply by TheRealMaN** [reply]
> ...I did...multiple times....in the browser....it was showing cached results 🤦‍♂️
> 
> This is working perfectly now, thanks!

Noob question) Can you share a working example? Or say what code to put in what file? I'm stuck(

**Reply by vitalijalbu** [reply]
> > ...I did...multiple times....in the browser....it was showing cached results 🤦‍♂️
> > This is working perfectly now, thanks!
> 
> Noob question) Can you share a working example? Or say what code to put in what file? I'm stuck(

hi, me too.. did u found a solution?

**Reply by ryanmitchell** [reply]
You can bind Statamic\Http\Resources\API\AssetResource class in a service provider and override the toArray() method to return the data you want included.
